lib/puppetx/puppetlabs/strings/yard/templates/default/html_helper.rb

```diff
@@ -0,0 +1,57 @@
+# A class containing helper methods to aid the generation of HTML
+# given formatted data
+class HTMLHelper
+
+  # Generates the HTML to format the relevant data about return values
+  def generate_return_types(types, desc = nil)
+    result = []
+
+    result << "(<span class=\"type\"><tt>" << types.join(", ") << "</tt></span>)"
+
+    if !desc.nil?
+      result << "- <div class=\"inline\"><p>#{desc}</p></div>"
+    end
+
+    result.join
+  end
+
+  # Generates the HTML to format the relevant data about parameters
+  def generate_parameters(params)
+    result = []
+
+    params.each do |param|
+      result << "<li>"
+
+      if !param[:exists?]
+        result << "<strike>"
+      end
+
+      result << "<span class=\"name\">#{param[:name]} </span>"
+      result << "<span class=\"type\">"
+
+      if param[:types]
+        result << "(" << "<tt>" << param[:types].join(", ") << "</tt>" << ")"
+      else
+        result << "(<tt>TBD</tt>)"
```

Comment:
3x functions will never have type info per parameter, they only describe the arity. So for those it will be very irritating to see TBD. We cannot really say anything - if we use `Any` it may be wrong (the function may only accept something else - there is no way to know). So for 3x functions I think we should leave this blank. In other cases 'TBD' is good enough for now - someone may want to change it later to 'unspecified' or something like that. We could run this by UX later, and ask for feedback.